cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -1101,18 +1101,30 @@ class VirtualMemoryResourceOptions:
     """A configuration object for the VirtualMemoryResource
        Stores configuration information which tells the resource how to use the CUDA VMM APIs
 
-    Args:
+    Parameters
+    ----------
+        allocation_type: VirtualMemoryAllocationTypeT
+            Controls the type of allocation.
+        location_type: VirtualMemoryLocationTypeT
+            Controls the location of the allocation.
         handle_type: Export handle type for the physical allocation. Use
             CU_MEM_HANDLE_TYPE_POSIX_FILE_DESCRIPTOR on Linux if you plan to
             import/export the allocation (required for cuMemRetainAllocationHandle).
             Use CU_MEM_HANDLE_TYPE_NONE if you don't need an exportable handle.
-        gpu_direct_rdma: Hint that the allocation should be GDR-capable (if supported).
-        granularity: 'recommended' or 'minimum'. Controls granularity query and size rounding.
-        addr_hint: A (optional) virtual address hint to try to reserve at. 0 -> let CUDA choose.
-        addr_align: Alignment for the VA reservation. If None, use the queried granularity.
-        peers: Extra device IDs that should be granted access in addition to `device`.
-        self_access: Access flags for the owning device ('rw', 'r', or 'none').
-        peer_access: Access flags for peers ('rw' or 'r').
+        gpu_direct_rdma: bool
+            Hint that the allocation should be GDR-capable (if supported).
+        granularity: VirtualMemoryGranularityT
+            Controls granularity query and size rounding.
+        addr_hint: int
+            A (optional) virtual address hint to try to reserve at. 0 -> let CUDA choose.
+        addr_align: int
+            Alignment for the VA reservation. If None, use the queried granularity.
+        peers: Iterable[int]
+            Extra device IDs that should be granted access in addition to `device`.
+        self_access: VirtualMemoryAccessTypeT
+            Access flags for the owning device ('rw' or 'r').
+        peer_access: VirtualMemoryAccessTypeT
+            Access flags for peers ('rw', 'r', or 'none').
```

Comment:
```suggestion
            Access flags for peers.
```
btw we should replace the string `"none"` by Python `None`. It was probably a miss from the previous code review. I don't think this would affect many users, so let's break it asap even if it is technically a breaking change.